stream-chat-android-docs/src/main/java/io/getstream/chat/docs/java/ui/guides/ImplementingOwnCapabilities.java

```diff
@@ -0,0 +1,42 @@
+package io.getstream.chat.docs.java.ui.guides;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+
+import io.getstream.chat.android.client.models.ChannelCapabilities;
+import io.getstream.chat.android.ui.common.state.messages.MessageMode;
+import io.getstream.chat.android.ui.common.state.messages.composer.MessageComposerState;
+import io.getstream.chat.android.ui.feature.messages.composer.MessageComposerView;
+
+/**
+ * [Implementing Own Capabilities](https://getstream.io/chat/docs/sdk/android/ui/guides/implementing-own-capabilities/)
+ */
+public class ImplementingOwnCapabilities {
+
+    private MessageComposerView messageComposerView;
+
+    public void customCapabilities() {
+        Set<String> customOwnCapabilities = new HashSet<>();
+        customOwnCapabilities.add(ChannelCapabilities.SEND_MESSAGE);
+        customOwnCapabilities.add(ChannelCapabilities.SEND_LINKS);
+        customOwnCapabilities.add(ChannelCapabilities.UPLOAD_FILE);
+        customOwnCapabilities.add(ChannelCapabilities.SEND_TYPING_EVENTS);
+
+        MessageComposerState messageComposerState = new MessageComposerState(
+                "",
+                Collections.emptyList(),
+                null,
+                Collections.emptyList(),
+                Collections.emptyList(),
+                Collections.emptyList(),
+                0,
+                MessageMode.Normal.INSTANCE,
+                false,
+                customOwnCapabilities,
+                false
+        );
```

Comment:
It's not related to this PR, but It would be better to add the `@JvmOverloads` annotation to the `MessageComposerState` data class and make this simple 😄

Reply:
Well, we need to pass the `ownCapabilities` parameter, which will still require us to use 10 arg constructor 😞

Reply:
@JvmOverloads won't help in that case. The only thing we could do is to introduce `Builders` but we should consider it as a future improvement